src/model.cpp

```diff
@@ -130,9 +130,10 @@ Status Model::addVersion(const ModelConfig& config) {
     return StatusCode::OK;
 }
 
-Status Model::addVersions(std::shared_ptr<model_versions_t> versionsToStart, ovms::ModelConfig& config, std::shared_ptr<FileSystem>& fs) {
+Status Model::addVersions(std::shared_ptr<model_versions_t> versionsToStart, ovms::ModelConfig& config, std::shared_ptr<FileSystem>& fs, std::shared_ptr<model_versions_t> versionsFailed) {
     Status result = StatusCode::OK;
     downloadModels(fs, config, versionsToStart);
+    versionsFailed->clear();
```

Comment:
In situation when we have many corrupted versions to add there might be many iterations in the loop trying to add new versions. The clear operation is to avoid removing previously failed versions from availableVersions. It would be duplicated because they were removed from available versions earlier. In means versionFailed should include just the versions failed in the current addVersions call.